sdks/jreleaser-mavencentral-java-sdk/src/main/java/org/jreleaser/sdk/mavencentral/MavenCentral.java

```diff
@@ -221,7 +221,14 @@ public <R> R retry(CheckedPredicate<R> stopFunction, CheckedSupplier<R> retriabl
             final int maxAttempts = maxRetries + 1;
 
             RetryPolicy<R> policy = RetryPolicy.<R>builder()
-                .handle(IllegalStateException.class, MavenCentralAPIException.class)
+                .handle(IllegalStateException.class)
+                .handleIf(exception -> {
+                    if (!(exception instanceof MavenCentralAPIException)) {
+                        return false;
+                    }
+
+                    return !((MavenCentralAPIException) exception).isUnauthorized();
```

Comment:
Given that the retry mechanism is _only_ used in the waitForStatus(), it seemed like the minimal possible change

Reply:
It looks OK though I rather invert the first condition and handle the MCE if there's a match.